### PR TITLE
Fixes variable scope

### PR DESCRIPTION
**Pull request by wleoncio**
Fixes variable scope issues on a few SMC functions.

I'm adding @anjastein as a reviewer ASAP to make sure the proposed assignments are correct, but maybe @osorensen can also tell if they look proper.

### CODE REVIEW

File: R/smc_post_processing_functions.R [hunk 1]
```diff
@@ -429,6 +431,7 @@ compute_posterior_intervals_rho <- function(output, nmc, burnin, verbose=FALSE)
 #'
 compute_rho_consensus <- function(output, nmc, burnin, C, type, verbose=FALSE) {
   n_items <- dim(output)[2]
+  N <- dim(output)[1]
   smc_plot <- smc_processing(output = output)
 
   iteration <- array(rep((1:N), n_items)) # ASK: what is N? It is out of scope. Gather from other objects or add as argument?
```
Comment (anjastein) [inline R/smc_post_processing_functions.R:437]: This is a typo - my mistake! I think N needs to be replaced with nmc.
N is the number of particles in SMC, but I use this function when running BayesMallows MCMC and SMC.
So when I run SMC and use this function I use the following input compute_rho_consensus(.... nmc = N ...).

File: R/smc_post_processing_functions.R [hunk 2]
```diff
@@ -429,6 +431,7 @@ compute_posterior_intervals_rho <- function(output, nmc, burnin, verbose=FALSE)
 #'
 compute_rho_consensus <- function(output, nmc, burnin, C, type, verbose=FALSE) {
   n_items <- dim(output)[2]
+  N <- dim(output)[1]
```
Comment (anjastein) [inline R/smc_post_processing_functions.R:434]: N is the input variable nmc.
When using this function to analyse SMC, I set nmc = N.

**Review by anjastein — COMMENTED** [review]
I'm happy with setting n_items = ncol(df).

The scope error in compute_rho_consensus is a typo - my mistake! I think N needs to be replaced with nmc. 
N is the number of particles in SMC, but I use this function to analyse output from BayesMallows MCMC and SMC. So when I run SMC and use this function, I input compute_rho_consensus(.... nmc = N ...).

**Comment by osorensen** [comment]
Was I too quick in merging this PR, then?

File: R/smc_post_processing_functions.R (hunk 2 above)
Comment (wleoncio) [reply R/smc_post_processing_functions.R:434]: OK, so I'll change this line to `N <- nmc`. A bit redundant, but best for now IMO.

**Comment by wleoncio** [comment]
> Was I too quick in merging this PR, then?

No worries, Øystein, I'll restore the branch and make the changes to comply with @anjastein's comments.

File: R/smc_post_processing_functions.R (hunk 1 above)
Comment (wleoncio) [reply R/smc_post_processing_functions.R:437]: Resolved by [comment below](https://github.com/ocbe-uio/BayesMallows/pull/79#discussion_r609680392).

**Comment by wleoncio** [comment]
Review addressed on commit 52a405cc926a287ea79719e2829c8be9e2d27b7d. I was going to revert the PR but it just felt too bureaucratic for an agreed one-liner, so I went ahead and merged it with `smc-mallows`.

**Pull request merged**